docs/watcher/README.md

```diff
@@ -11,10 +11,10 @@ The Watcher currently supports the following types:
 
 ## Result Grouping
 
-The Watcher uses annotations to automatically detect and group related Records into the same Result. The following annotations are recognized (in order of precedence):
+The Watcher uses Object data to automatically detect and group related Records into the same Result. The following data is checked (in order of precedence):
 
-- `results.tekton.dev/result`
-- `triggers.tekton.dev/triggers-eventid`
-- `tekton.dev/pipelineRun`
+- `results.tekton.dev/result` annotation. This should correspond to the full `Result.name` identifier (e.g. `foo/results/bar`).
+- `triggers.tekton.dev/triggers-eventid` annotation (this is generated from Objects created via [Tekton Triggers](https://github.com/tektoncd/triggers))
+- An [OwnerReference](https://kubernetes.io/docs/concepts/workloads/controllers/garbage-collection/#owners-and-dependents) to a PipelineRun.
```

Comment:
minor: think TaskRuns created by a PipelineRun will also have a `tekton.dev/pipelineRun` label added that could be used instead (might make this a tad easier since we are already using labels/annotations for the other two).

Reply:
Yeah I thought about doing this, but decided against it because using OwnerReference gives us direct access to the PipelineRun UID, avoiding a additional API call and any race conflicts that might arise from that.